tests/Kernel/DateRangeFieldTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme\Kernel;
+
+use Drupal\datetime_range\Plugin\Field\FieldType\DateRangeItem;
+use Drupal\entity_test\Entity\EntityTest;
+use Drupal\oe_theme\ValueObject\DateValueObject;
+use Drupal\Tests\datetime_range\Kernel\DateRangeItemTest;
+use Drupal\Tests\oe_theme\Traits\RenderTrait;
+use Symfony\Component\Yaml\Yaml;
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+
+/**
+ * Test date value object with datetime_range field type.
+ */
+class DateRangeFieldTest extends DateRangeItemTest {
```

Comment:
Since the `DateRangeFieldTest` implements some tests, we should try to extend from `FieldKernelTestBase` so we don't run unrelated tests.

Reply:
I think we should also rename the test, we are testing the date value objects, maybe follow the same approach as the unit tests folder and have a separate folder for the value objects and have a test there? Maybe we can merge also the test from 1379 in it (not sure it's a good idea).

Reply:
Done. But required review and take a look again on new structure of classes.
I moved testing class to new ValueObject directory with renaming. Naming of classes based on wish to do more `readable`. So will be like `ValueObject/FromDateTimeRangeFieldTest` and `ValueObject/FromDateTimeFieldTestBase`.

Reply:
We discussed again together and we are gonna go for a small refactor on this. We are also gonna drop the rendering test as using a different factory method won't change the way a date value object is gonna be rendered.